pixels/endpoints/general.py

```diff
@@ -0,0 +1,175 @@
+    return Response(bytes(await request.state.canvas.get_pixels()),
+                    media_type="application/octet-stream")
+
+
+@router.get("/pixel", response_model=Pixel)
+@ratelimits.UserRedis(
+    requests=Ratelimits.GET_PIXEL_AMOUNT,
+    time_unit=Ratelimits.GET_PIXEL_RATE_LIMIT,
+    cooldown=Ratelimits.GET_PIXEL_RATE_COOLDOWN
+)
+async def get_pixel(x: int, y: int, request: Request) -> Pixel:
+    """
+    Get a single pixel given the x and y coordinates.
+
+    This endpoint requires an authentication token.
+    See [this page](https://pixels.pythondiscord.com/info/authentication)
+    for how to authenticate with the API.
+
+    #### Example Python Script
+    ```py
+    from dotenv import load_dotenv
+    from os import getenv
+    import requests
+
+    load_dotenv(".env")
+
+    token = getenv("TOKEN")
+
+    headers = {"Authorization": f"Bearer {token}"}
+
+    r = requests.get(
+        "https://pixels.pythondiscord.com/get_pixel",
+        headers=headers,
+        params={  # Note: we're using query string parameters to define the coordinates, not the JSON body.
+            "x": 87,
+            "y": 69
+        }
+    )
+
+    print("Here's the colour of the pixel:", r.json()["rgb"])
+    ```
+    """
+    if x >= Sizes.WIDTH or y >= Sizes.HEIGHT:
+        raise HTTPException(400, "Pixel is out of the canvas bounds.")
+    pixel_data = await request.state.canvas.get_pixel(x, y)
+
+    return Pixel(x=x, y=y, rgb=''.join(f"{x:02x}" for x in pixel_data))
+
+
+@router.put("/pixel", response_model=Message)
+@ratelimits.UserRedis(
+    requests=Ratelimits.PUT_PIXEL_AMOUNT,
+    time_unit=Ratelimits.PUT_PIXEL_RATE_LIMIT,
+    cooldown=Ratelimits.PUT_PIXEL_RATE_COOLDOWN
+)
+async def put_pixel(request: Request, pixel: Pixel) -> Message:
+    """
+    Override the pixel at the specified position with the specified color.
+
+    This endpoint requires an authentication token.
+    See [this page](https://pixels.pythondiscord.com/info/authentication)
+    for how to authenticate with the API.
+
+    #### Example Python Script
+    ```py
+    from dotenv import load_dotenv
+    from os import getenv
+    import requests
+
+    load_dotenv(".env")
+
+    token = getenv("TOKEN")
+
+    headers = {"Authorization": f"Bearer {token}"}
+    data = {
+      "x": 80,
+      "y": 45,
+      "rgb": "00FF00"
+    }
+    r = requests.post(  # remember, this is a POST method not a GET method.
+        "https://pixels.pythondiscord.com/set_pixel",
+        json=data,
+        headers=headers,
+    )
+    payload = r.json()
+
+    print(f"We got a message back! {payload['message']}")
+    ```
+    """
+    log.info(f"{request.state.user_id} is setting {pixel.x}, {pixel.y} to {pixel.rgb}")
+    await request.state.canvas.set_pixel(request.state.db_conn, pixel.x, pixel.y, pixel.rgb, request.state.user_id)
+    return Message(message=f"added pixel at x={pixel.x},y={pixel.y} of color {pixel.rgb}")
```

Comment:
```suggestion
    return Message(message=f"Set pixel at x={pixel.x},y={pixel.y} to color {pixel.rgb}")
```
Should this have a trailing period as well?